src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -1075,6 +1078,67 @@ func (c controller) logClusterOperatorsStatus() {
 	}
 }
 
+// This is temporary function that should validate router state only in case of failure
+// It will patch router to add access logs
+// It will run http router health check to see if router is healthy on host network
+func (c controller) logRouterStatus() {
+	if !c.Status.HasError() || c.HighAvailabilityMode != models.ClusterHighAvailabilityModeNone {
+		return
+	}
+	c.log.Infof("Start checking router status")
+	var cl *models.Cluster
+	var err error
+	patched := false
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr, Timeout: dnsValidationTimeout}
+	localIp, _ := utils.GetOutboundIP()
+	if localIp == "" {
+		localIp = "localhost"
+	}
+
+	var consoleUrl string
+	// run every 30 seconds router check and log the result
+	go utils.WaitForPredicate(WaitTimeout, SummaryLogsPeriod, func() bool {
+		if consoleUrl == "" {
+			cl, err = c.ic.GetCluster(context.Background(), false)
+			if err != nil {
+				c.log.WithError(err).Errorf("Failed to get cluster %s from assisted-service", c.ClusterID)
+				return false
+			}
+		}
+
+		consoleUrl = fmt.Sprintf("https://console-openshift-console.apps.%s.%s/health", cl.Name, cl.BaseDNSDomain)
+		r, err := client.Get(consoleUrl)
+		if err != nil {
+			c.log.WithError(err).Warning("Failed to reach console")
+		} else {
+			c.log.Infof("console url status %s", r.Status)
+		}
+
+		url := fmt.Sprintf("http://%s/_______internal_router_healthz", localIp)
+		_, err = client.Get(url)
+		if err != nil {
+			c.log.WithError(err).Warning("Failed to reach internal router health")
+		} else {
+			c.log.Infof("route internal health status %s", r.Status)
+		}
+
+		if !patched {
```

Comment:
Why have the patch in the predicate? why not do it once in case the predicate failed

Reply:
cause i want to check status before the check and in that case i need to move check from predicate.

Reply:
so it doesn't actually matter. Don't see reason to move it to blocking part of the function